### PR TITLE
.NET: test: add AgentResponse edge case coverage

### PR DESCRIPTION
**Pull request by challaravinath**
### Motivation and Context

  AgentResponse had no tests covering null ChatResponse construction,
  Text property behavior with non-text content, Text after Messages
  is set to null, or message order preservation in ToAgentResponseUpdates.
  These are gaps in critical response handling paths.
-->

### Description

  - Add test verifying null ChatResponse constructor throws ArgumentNullException
  - Add test verifying Text returns empty string when all content is non-text
  - Add test verifying Text returns empty string after Messages is set to null
  - Add test verifying ToAgentResponseUpdates preserves message order across multiple messages

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
 - [x] The PR follows the Contribution Guidelines
 - [x] All unit tests pass, and I have added new tests where possible
 - No breaking change test only, no production code modified

### CODE REVIEW

**Review by lokitoth — APPROVED** [review]

File: dotnet/tests/Microsoft.Agents.AI.Abstractions.UnitTests/AgentResponseTests.cs [hunk 1]
```diff
@@ -287,4 +287,67 @@ public void ToAgentResponseUpdatesWithAdditionalPropertiesOnlyProducesSingleUpda
         Assert.NotNull(update.AdditionalProperties);
         Assert.Equal("value", update.AdditionalProperties!["key"]);
     }
+
+    [Fact]
+    public void Constructor_WithNullChatResponse_ThrowsArgumentNullException()
+    {
+        // Act & Assert
+        Assert.Throws<ArgumentNullException>(() => new AgentResponse((ChatResponse)null!));
```
[the remark this answers is not in the document]
Comment (challaravinath) [reply dotnet/tests/Microsoft.Agents.AI.Abstractions.UnitTests/AgentResponseTests.cs]: Implemented the Recommended Change

**Review by rogerbarreto — APPROVED** [review]
